client/state/sharing/keyring/selectors.js

```diff
@@ -44,6 +44,19 @@ export const getKeyringConnectionsByName = createSelector(
 	state => [ state.sharing.keyring.items ]
 );
 
+/**
+ * Returns an array of broken keyring connection objects for a specified service.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {String} service Service slug.
+ * @return {Array}         Keyring connections, if known.
+ */
+export function getBrokenKeyringConnectionsByName( state, service ) {
+	return filter( getKeyringConnectionsByName( state, service ), {
+		status: 'broken',
+	} );
+}
+
```

Comment:
The others selectors in this file have tests in `client/state/sharing/keyring/test/selectors.js` -- would be good to add one for this new selector.